usbhost.h

```diff
@@ -503,9 +503,28 @@ int8_t MAX3421e< SPI_SS, INTR >::Init(int mseconds) {
 /* probe bus to determine device presence and speed and switch host to this speed */
 template< typename SPI_SS, typename INTR >
 void MAX3421e< SPI_SS, INTR >::busprobe() {
+        static uint8_t prev_bus = -1;
         uint8_t bus_sample;
+
+        if (regRd(rHIRQ) & bmCONDETIRQ) {
+            // device plug/unplug is detected
+            // clear CONDETIRQ
+            regWr(rHIRQ, bmCONDETIRQ);
+        } else {
+            // check current bus state when no device(SE0),
+            // otherwise transient state will be read due to bus activity.
+            if (vbusState != SE0) return;
+
+            // read current bus state
+            regWr(rHCTL, bmSAMPLEBUS);
```

Comment:
Shouldn't you wait for the operation to finish like how it is done elsewhere in the code?

I.e:

```C++
while(!(regRd(rHCTL) & bmSAMPLEBUS)); //wait for sample operation to finish
```

Reply:
That code seems to use contrary logic according to MAX3421E reference?
From my tests the waiting code does nothing and is not needed in fact.

The chip reference(AN3785.pdf p.52) says:
> The CPU sets the SAMPLEBUS bit to instruct the SIE to sample the state of the D+ and D-
lines. The SIE clears the SAMPLEBUS bit when it completes the operation.

But the `SAMPLEBUS` bit shows strange behaviour there in fact.

I never seen 'S' with code below.

```while (!(regRd(rHCTL) & bmSAMPLEBUS)) { Serial.print("S "); };```

While this code is repeated forever for some reason.

```while (regRd(rHCTL) & bmSAMPLEBUS) { Serial.print("S "); }; ```

Reply:
I guess the SAMPLEBUS bit is cleared in between the two calls.

Reply:
With some more tests the SAMPLEBUS bits doesn't seems to be cleared until bus state is changed but not always. This is clearly not what the reference describes, we can't use the bit to wait for sampling bus state anyway.

And simply asserting SAMPLEBUS everytime before reading J/KSTATUS is good practice seemingly. This is what MaximUSBLab10 code does.

My code works in my tests at least so far.